integration-tests/tests/src/tests/objects.ts

```diff
@@ -77,6 +77,7 @@ describe("Realm objects", () => {
       });
 
       const john = realm.objectForPrimaryKey<IPersonWithId>(PersonSchemaWithId.name, _id);
+      if (!john) throw new Error("Object not found");
```

Comment:
Can we `expect(john).to.not.be(undefined)`, or does that defeat the purpose of this PR?

Reply:
We could, but this wouldn't satisfy Typescript so we'd have to do non-null assertions or whatever. I thought that as its an edge case (like should never happen), throwing is the least lines of code required to both fail the test and keep TS happy, but the test error may be a little less clear if you did hit the edge case

Reply:
Ok...that makes sense.  We need to keep TS happy 🍼